materials/views.py

```diff
@@ -1,42 +1,62 @@
-from rest_framework.generics import (CreateAPIView, DestroyAPIView,
-                                     ListAPIView, RetrieveAPIView,
-                                     UpdateAPIView)
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.viewsets import ModelViewSet
 
 from materials.models import Course, Lesson
-from materials.serializers import CourseSerializer, LessonSerializer
+from materials.serializers import CourseDetailSerializer, CourseSerializer, LessonSerializer
+from users.permissions import IsModers, IsOwner
 
-"""CRUD для модели курс по Viewsets"""
 
-
-class CourseViewSets(ModelViewSet):
+class CourseViewSet(ModelViewSet):
     queryset = Course.objects.all()
-    serializer_class = CourseSerializer
 
+    def get_permissions(self):
+        if self.action == "create":
+            self.permission_classes = (~IsModers,)
+        elif self.action == "destroy":
+            self.permission_classes = (~IsModers | IsOwner,)
+        else:
+            self.permission_classes = (IsModers | IsOwner,)
+        return super().get_permissions()
 
-"""CRUD для модели урока по Generic"""
+    def perform_create(self, serializer):
+        course = serializer.save(owner=self.request.user)
+        course.save()
 
+    def get_queryset(self):
+        if not IsModers().has_permission(self.request, self):
+            return Course.objects.filter(owner=self.request.user)
+        return Course.objects.all()
 
-class LessonCreateApiView(CreateAPIView):
-    queryset = Lesson.objects.all()
+
+class LessonCreateAPIView(generics.CreateAPIView):
     serializer_class = LessonSerializer
+    permission_classes = (~IsModers, IsAuthenticated)
 
+    def perform_create(self, serializer):
+        lesson = serializer.save()
```

Comment:
Достаточно одной строки
```suggestion
        serializer.save(owner=self.request.user)
```